web3/txmanager/txsend.go

```diff
@@ -0,0 +1,468 @@
+			continue
+		}
+		// Transaction is stuck - attempt replacement
+		log.Warnw("stuck transaction detected",
+			"id", fmt.Sprintf("%x", ptx.ID),
+			"nonce", nonce,
+			"age", txAge,
+			"hash", ptx.Hash.Hex(),
+			"retries", ptx.RetryCount)
+		if err := tm.speedUpTx(ctx, ptx); err != nil {
+			log.Errorw(err, fmt.Sprintf("failed to speed up transaction for nonce %d", nonce))
+		}
+	}
+
+	return nil
+}
+
+// speedUpTx attempts to speed up a stuck transaction by resending with higher
+// fees. If max retries are reached, it cancels the transaction if possible.
+// It returns an error if the operation fails.
+func (tm *TxManager) speedUpTx(ctx context.Context, ptx *PendingTransaction) error {
+	if ptx.RetryCount >= tm.config.MaxRetries {
+		// Max retries reached, cancel transaction
+		return tm.cancelTx(ctx, ptx)
+	}
+	// Check if transaction was already mined
+	onChainNonce, err := tm.lastOnChainNonce(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get last on-chain nonce: %w", err)
+	}
+	if onChainNonce > ptx.Nonce {
+		// pending nonce already advanced on-chain, skipping speed-up
+		tm.updateNonceTracking(ptx.Nonce)
+		return nil
+	}
+	// Calculate new gas price
+	increaseFactor := big.NewInt(int64(100 + tm.config.FeeIncreasePercent))
+	newGasPrice := new(big.Int).Mul(ptx.OriginalGasPrice, increaseFactor)
+	newGasPrice.Div(newGasPrice, big.NewInt(100))
+	// Cap at max gas price
+	if newGasPrice.Cmp(tm.config.MaxGasPriceGwei) > 0 {
+		newGasPrice = new(big.Int).Set(tm.config.MaxGasPriceGwei)
+	}
+	var newTx *gtypes.Transaction
+	if ptx.IsBlob {
+		// For blob transactions, also increase blob gas fee
+		newBlobFee := new(big.Int).Mul(ptx.OriginalBlobFee, increaseFactor)
+		newBlobFee.Div(newBlobFee, big.NewInt(100))
+		// Check if sidecar is available for rebuilding
+		if ptx.BlobSidecar != nil {
+			// Rebuild blob transaction with higher fees
+			newTx, err = tm.rebuildBlobTx(ctx, ptx, newGasPrice, newBlobFee)
+			if err != nil {
+				return fmt.Errorf("cannot rebuild blob transaction: %w", err)
+			}
+		} else {
+			// CRITICAL: Blob transactions cannot be cancelled. They can only
+			// be replaced with another blob transaction using the same blob
+			// data. Without the sidecar, we cannot create a replacement, so
+			// the transaction is permanently stuck.
+			return fmt.Errorf("blob transaction stuck without recovery option")
+		}
+	} else {
+		newTx, err = tm.rebuildRegularTx(ctx, ptx, newGasPrice)
+		if err != nil {
+			return fmt.Errorf("failed to rebuild transaction: %w", err)
+		}
+	}
+	// Send replacement
+	if err := tm.cli.SendTransaction(ctx, newTx); err != nil {
+		if isNonceError(err) {
+			// replacement unnecessary, nonce already advanced
+			tm.updateNonceTracking(ptx.Nonce)
+			return nil
+		}
+		return fmt.Errorf("failed to send replacement: %w", err)
+	}
+	// Update tracking
+	ptx.Hash = newTx.Hash()
+	ptx.RetryCount++
+	ptx.Timestamp = time.Now()
+	ptx.OriginalGasPrice = newGasPrice
+	if ptx.IsBlob {
```

Comment:
Consider putting all this logic within the switch/case, instead of inline splitting operations depending on a specific case.